File: server/server/Models/Entities/UserBattle.cs
namespace server.Models.Entities;

public class UserBattle
{
    public int Id { get; set; }
    public int Punctuation { get; set; } = 0;
    public long TimePlayed { get; set; } = 0;
    public int UserId { get; set; }
    public int BattleResultId { get; set; } = 1;
    public bool Receiver { get; set; } = false;
    public User User { get; set; } = null;
    public BattleResult BattleResult { get; set; } = null;
    public bool IsHost { get; set; } = false;
    public Character Character { get; set; } = null;
    public int CharacterId { get; set; } = 1;

    public int Position { get; set; } = 0;
    public int TotalKills { get; set; } = 0;
}

File: web/chokart/src/app/models/user-battle.ts
import { Character } from './character';

export interface UserBattle {
  id: number;
  userId: number;
  punctuation: number;
  timePlayed: number;
  battleResultId: number;
  isHost: boolean;
  character: Character;
  characterId: number;
  position: number;
  totalKills: number;
}
